core/src/main/java/com/linecorp/armeria/client/retry/BackoffSpec.java

```diff
@@ -0,0 +1,215 @@
+                      initialDelayMillis, maxDelayMillis, multiplier);
+        exponentialConfigured = true;
+
+        final String[] values = exponentialValues.split(SPLIT_VALUES);
+        checkArgument(values.length < 4,
+                      "the values '%s' for exponential parameters should be less than 4.", exponentialValues);
+
+        long initialDelayMillis = Math.abs(parseLong(values[0].trim(), DEFAULT_INITIAL_DELAY_MILLIS));
+        long maxDelayMillis =
+                values.length >= 2 ? Math.abs(parseLong(values[1].trim(), DEFAULT_MAX_DELAY_MILLIS))
+                                   : DEFAULT_MAX_DELAY_MILLIS;
+        if (initialDelayMillis < maxDelayMillis) {
+            this.initialDelayMillis = initialDelayMillis;
+            this.maxDelayMillis = maxDelayMillis;
+        } else {
+            this.initialDelayMillis = maxDelayMillis;
+            this.maxDelayMillis = initialDelayMillis;
+        }
+
+        multiplier = values.length == 3 ? parseDouble(values[2].trim(), DEFAULT_MULTIPLIER)
+                                        : DEFAULT_MULTIPLIER;
+    }
+
+    private void jitter(String jitterValues) {
+        checkArgument(!jitterConfigured,
+                      "jitter parameters are already set. minJitterRate: %.3f, maxJitterRate: %.3f",
+                      minJitterRate, maxJitterRate);
+        jitterConfigured = true;
+
+        final String[] values = jitterValues.split(SPLIT_VALUES);
+        checkArgument(values.length < 3,
+                      "the values '%s' for jitter parameters should be less than 4.", jitterValues);
+
+        if (values.length == 1) {
+            double jitterRate = parseDouble(values[0].trim(), DEFAULT_MAX_JITTER_RATE);
+            checkArgument(0.0 <= jitterRate && jitterRate <= 1.0,
+                          "jitterRate: %f (expected: >= 0.0 and <= 1.0)", jitterRate);
+            minJitterRate = jitterRate * -1;
+            maxJitterRate = jitterRate;
+        } else {
+            minJitterRate = parseDouble(values[0].trim(), DEFAULT_MIN_JITTER_RATE);
+            maxJitterRate = parseDouble(values[1].trim(), DEFAULT_MAX_JITTER_RATE);
+            if (minJitterRate > maxJitterRate) {
+                double temp = minJitterRate;
+                minJitterRate = maxJitterRate;
+                maxJitterRate = temp;
+            }
+        }
+    }
+
+    private void maxAttempts(String value) {
+        checkArgument(!maxAttemptsConfigured, "maxAttempts parameters is already set. maxAttempts: %d",
+                      maxAttempts);
+        maxAttemptsConfigured = true;
+        maxAttempts = parseInt(value, DEFAULT_MAX_ATTEMPTS);
+        checkArgument(maxAttempts > 0, "maxAttempts: %d (expected: > 0)", maxAttempts);
+    }
+
+    private static long parseLong(String value, Long defaultValue) {
+        try {
+            return Long.parseLong(value);
+        } catch (NumberFormatException e) {
+            return defaultValue;
```

Comment:
What do you think about raising an exception instead of using the default parameter value? Otherwise a user may think he or she specified the correct value.